pkg/ssh/ssh.go

```diff
@@ -18,9 +20,56 @@ import (
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 )
 
+// LoadKey loads the SSH public key specified in NodeGroupSSH and returns it. The key should be specified
+// in only one way: by name (for a key existing in EC2), by path (for a key in a local file)
+// or by its contents (in the config-file). It also assumes that if ssh is enabled (SSH.Allow
+// == true) then one key was specified
+func LoadKey(sshConfig *api.NodeGroupSSH, clusterName, nodeGroupName string, ec2API ec2iface.EC2API) (string, error) {
```

Comment:
My initial  idea when I designed this logic was that the ssh package would only do the low level stuff, talking to aws apis.

This loading is closer to the UI, because it's very specific to how a user can specify ssh keys in the API schema. While the ssh package is UI agnostic, and just knows how to do certain things like loading a key by filename, but it doesn't know that a user can specify it as a key, name or content, that's a UI decision -> `ctl` layer.

I'm not completely against moving it, but let's decide on the strategy we want to follow since we will be doing more refactors for the versioning. @marccarre what do you think?

Reply:
I don't have a strong opinion on this _at the moment_. I agree we generally shouldn't mix higher-level logic with lower-level logic. However, that being said, it looks like other methods of this file were also depending on `pkg/apis`' types, e.g. `api.ClusterProvider`, so it doesn't _feel_ too different.

Reply:
After discussion with @martina-if , I have moved the low-level SSH functions to package `ssh/client` and kept LoadKey in the top-level `ssh` package.